lahja/asyncio/endpoint.py

```diff
@@ -360,16 +382,77 @@ def run(self, *args, **kwargs):  # type: ignore
     async def start(self) -> None:
         if self.is_running:
             raise RuntimeError(f"Endpoint {self.name} is already running")
+
         self._receiving_loop_running = asyncio.Event()
         self._receiving_queue = asyncio.Queue()
 
+        self._subscription_updates_running = asyncio.Event()
+        self._subscription_updates_condition = asyncio.Condition()
+        self._subscription_updates_queue = asyncio.Queue()
+
         self._running = True
 
         self._endpoint_tasks.add(asyncio.ensure_future(self._connect_receiving_queue()))
+        self._endpoint_tasks.add(
+            asyncio.ensure_future(self._process_subscription_updates())
+        )
+        self._endpoint_tasks.add(
+            asyncio.ensure_future(self._process_subscription_updates_queue())
+        )
 
         await self._receiving_loop_running.wait()
+        await self._subscription_updates_running.wait()
+
         self.logger.debug("Endpoint[%s]: running", self.name)
 
+    async def _process_subscription_updates_queue(self) -> None:
+        while self.is_running:
+            await self._subscription_updates_queue.get()
+            async with self._subscription_updates_condition:
+                self._subscription_updates_condition.notify_all()
+
+    async def _process_subscription_updates(self) -> None:
+        self._subscription_updates_running.set()
+        async with self._subscription_updates_condition:
```

Comment:
@lithp Needed this to ensure we don't end up with concurrent subscription updates being sent out since it's possible that they arrive out of order in that case and then the other side will have potentially in-accurate subscription records.

TOOD: add a code comment explaining this.

Reply:
How would they arrive out of order? Domain sockets always deliver messages in order and `_notify_lock` ensures there's only one update per socket happening at any time.

Reply:
I started undoing this but then stopped as I'm inclined to leave it this way.  Here's my reasoning.

1. we end up needing to do subscription updates from synchronous methods with the addition of `subscribe_nowait` which seems to be a necessary API. 
2. I'm much more comfortable with that method doing a `Queue.put_nowait` than I am adding an additional call to `asyncio.ensure_future(self._notify_subscription_updates())`.
3. I think that the `asyncio.Condition` based approach is more appropriate and under that approach we still need a background process to listen for changes.  The various branches that I have in progress have made good use of that `Condition` API.

I'm fine unrolling this if you're still not convinced at the point where the `trio` based endpoint is being added to the library.